src/components/Alphabet/Letter/Letter.js

```diff
@@ -1,34 +1,62 @@
 import React from 'react';
 import styles from './Letter.css';
+import classnames from 'classnames';
 
-const checkLetter = (letter, id, onLetterClick) => {
-    const url = `/phrases/check`;
-    return fetch(url, {
-        headers: {
-            'Content-Type': 'application/json',
-            'Accept': 'application/json',
-        },
-        method: 'post',
-        body: JSON.stringify({
-            id,
-            letter,
+class Letter extends React.PureComponent {
+    state = {
+        isLetterCorrect: null,
+        usedLetter: false,
+    };
+
+    checkLetter = (letter, id) => {
+        const url = `/phrases/check`;
+        return fetch(url, {
+            headers: {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json',
+            },
+            method: 'post',
+            body: JSON.stringify({
+                id,
+                letter,
+            })
         })
-    })
-    .then(response => response.json())
-    .catch(error => {
-        console.error(error);
-    })
-    .then(data => {
-        onLetterClick(data);
-    });
-}
+        .then(response => response.json())
+        .catch(error => {
+            console.error(error);
+        })
+        .then(data => {
+            this.setState({
+                isLetterCorrect: data.failsCounter === this.props.failsCounter,
+            })
+            this.props.onLetterClick(data);
+        });
+    }
 
-const Letter = React.memo(({letter, gameID, onLetterClick}) => {
-    return(
-        <div className={styles.letter} onClick={checkLetter.bind(null, letter, gameID, onLetterClick)}>
-            {letter}
-        </div>
-    )
-});
+    onClickMethod = () => {
+        if (this.state.usedLetter) return;
+        this.setState ({
+            usedLetter: true,
+        })
+        const { letter, gameID } = this.props;
+        this.checkLetter(letter, gameID);
+    }
+
+    render() {
+        const { usedLetter, isLetterCorrect } = this.state;
+
+        const letterStyle = classnames(styles.letter, {
+            [styles.letterHover]: !usedLetter,
+        }, usedLetter && isLetterCorrect !== null && {
+            [styles.correctLetter]: isLetterCorrect,
+            [styles.wrongLetter]: !isLetterCorrect,
+        });
```

Comment:
This looks weirdly. In general:
- instead of 'isLetterCorrect', have a variable called 'letterStatus' which might be 'LETTER_STATUS_CORRECT' or 'LETTER_STATUS_INCORRECT'. Then check that instead of checking specifically for !== null,
- make 'hover' a default behaviour of letter instead of having separate class just for that, it's stupid,
- you can write that as:
```
const letterCorrectClassName = 
const letterStyle = classnames(styles.letter, {
    [styles.correct]: letterStatus === LETTER_STATUS_CORRECT,
    [styles.incorrect]: letterStatus === LETTER_STATUS_INCORRECT
});
```

Reply:
In general you're right. Now with pointer-events: none; that you have mentioned above is easier to manage hover without variable. I have just refactored code according to your points. I hope it will be good.